contrib/mesos/pkg/queue/historical.go

```diff
@@ -377,6 +377,19 @@ func (f *HistoricalFIFO) merge(id string, obj UniqueCopyable) (notifications []E
 	return
 }
 
+// Resync will touch all objects to put them into the processing queue
+func (f *HistoricalFIFO) Resync() error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	for id, entry := range f.items {
+		if entry.Is(ADD_EVENT|UPDATE_EVENT) {
+			f.queue = append(f.queue, id)
+		}
+	}
+	f.cond.Broadcast()
+	return nil
```

Comment:
Sure.
So basically, so far most of  of reflectors/informers etc. were periodically relisting objects from etcd. However, this doesn't make much sense. However, different controllers rely on the fact that they need to process all objects every X seconds/minutes. So instead of doing relist every such period, we are introducing resync operation, to avoid unnecessary operations on etcd.